mono/metadata/class.c

```diff
@@ -1025,7 +1025,8 @@ mono_class_inflate_generic_method_full_checked (MonoMethod *method, MonoClass *k
 	MonoMethodSignature *sig;
 	MonoGenericContext tmp_context;
 
-	error_init (error);
+	error_init (error); // for the assert
```

Comment:
If you are doing it for the assert, why not just have the assert? It would make it much clearer as whenever anybody in the future reads it (it's not very clear when hovering over the code that `error_init` does an assert).

Reply:
The init field is semi-private. What assert would I put here out in the wild? Maybe assert(error_is_initialized())? Ok.

Reply:
(or warn, yes)